### PR TITLE
Fix modules redirects with pre-defined filters

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description
This PR adds the functionality to add filters via URL.

### Test

- Go to Server Management > Rules
- Click on 'View alerts of this Rule' 
- Check the filter is applied after redirects
- Navigate to another modules and check that the filters disappear (prevents the error when the index pattern changes and the field doesn't exists)
 

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/03b7438e-e2cb-47bd-b3d3-b654aab9ba5a

### Check List
- [x] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by chantal-kelm — DISMISSED** [review]
CR: ✅ 
Test: ✅ 



https://github.com/wazuh/wazuh-dashboard-plugins/assets/99441266/fd98fe83-05d6-4cb8-be3c-8328746baf88

**Comment by asteriscos** [comment]
## Test :green_circle: 

Peek 2024-05-30 20-48

File: plugins/main/public/components/common/data-source/pattern/pattern-data-source-filter-manager.ts [hunk 1]
```diff
@@ -373,6 +390,141 @@ export class PatternDataSourceFilterManager
...
+                  query: value,
+                },
+              },
+            },
+          },
+          controlledBy,
+          true,
+        );
+      case 'exists':
+        // '{"meta":{"index":"wazuh-alerts-*","alias":null,"negate":false,"disabled":false,"type":"exists","key":"agent.id","value":"exists"},"exists":{"field":"agent.id"},"$state":{"store":"appState"}}'
```
Comment (asteriscos) [inline plugins/main/public/components/common/data-source/pattern/pattern-data-source-filter-manager.ts]: Remove the comment with the example please.

**Review by yenienserrano — APPROVED** [review]
LGTM


Uploading Screen Recording 2024-05-30 at 8.50.50 PM.mov…

**Review by asteriscos — APPROVED** [review]

**Pull request merged**